src/lib/random.js

```diff
@@ -340,6 +341,88 @@ var $builtinmodule = function (name) {
         return new Sk.builtin.float_(c);
     });
 
+    mod.triangular = new Sk.builtin.func(function (low, high, mode) {
+        Sk.builtin.pyCheckArgs("triangular", arguments, 2, 3);
+
+        var rnd, sample, swap;
+
+        low = Sk.builtin.asnum$(low);
+        high = Sk.builtin.asnum$(high);
+        if (low > high) {
+            swap = low;
+            low = high;
+            high = swap;
+        }
+        if (mode === undefined) {
+            mode = (high - low)/2.0;
+        } else {
+            mode = Sk.builtin.asnum$(mode);
+        }
+
+        // https://en.wikipedia.org/wiki/Triangular_distribution
+        rnd = myGenerator.genrand_res53();
+        if (rnd < (mode - low)/(high - low)) {
+            sample = low + Math.sqrt(rnd * (high - low) * (mode - low));
+        } else {
+            sample = high - Math.sqrt((1 - rnd) * (high - low) * (high - mode));
+        }
+
+        return new Sk.builtin.float_(sample);
+    });
+
+    var normalSample = function(mu, sigma) {
+        var r1, r2, u, v, s;
+
+        // Box-Muller transform
+        // (https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform)
+        // generates two independent samples from a Gaussian
+        // distribution. Return one of them and store the another one
+        // and return it next time.
+
+        if (nextNormalSample !== undefined) {
+            s = nextNormalSample;
+            nextNormalSample = undefined;
+        } else {
+            r1 = myGenerator.genrand_res53();
+            r2 = myGenerator.genrand_res53();
+            u = Math.sqrt(-2*Math.log(r1));
+            v = 2*Math.PI*r2;
+            s = u * Math.cos(v);
+            nextNormalSample = u * Math.sin(v);
+        }
+
+        return mu + sigma*s;
+    };
+    
+    mod.gauss = new Sk.builtin.func(function (mu, sigma) {
+        Sk.builtin.pyCheckArgs("gauss", arguments, 2, 2);
+
+        mu = Sk.builtin.asnum$(mu);
+        sigma = Sk.builtin.asnum$(sigma);
+
+        return new Sk.builtin.float_(normalSample(mu, sigma));
+    });
+
+    mod.normalvariate = mod.gauss;
```

Comment:
In CPython those two methods are a little bit different. `normalvariate` uses a different algorithm `Kinderman and Monahan` and is thread safe. The `gauss` is not thread-safe and may be faster. I think those differences do not really affect your implementation and the usage of skulpt. However, I think it would be better to drop a comment in the source.